File: src/main/java/com/itsci/mju/maebanjumpen/service/HousekeeperSkillService.java
package com.itsci.mju.maebanjumpen.service;

import com.itsci.mju.maebanjumpen.dto.HousekeeperDTO;
import com.itsci.mju.maebanjumpen.dto.HousekeeperSkillDTO;

import java.util.List;
import java.util.Optional;

public interface HousekeeperSkillService {
    List<HousekeeperDTO> getAllHousekeeperSkills();
    HousekeeperSkillDTO getHousekeeperSkillById(int id);
    HousekeeperSkillDTO saveHousekeeperSkill(HousekeeperSkillDTO housekeeperSkillDto);
    void deleteHousekeeperSkill(int id);
    Optional<HousekeeperSkillDTO> getSkillsByHousekeeperId(int housekeeperId);
    HousekeeperSkillDTO updateHousekeeperSkill(int id, HousekeeperSkillDTO skillDto);
    void updateSkillLevelAndHiresCompleted(Integer housekeeperId, Integer skillTypeId);
    Optional<HousekeeperSkillDTO> findByHousekeeperIdAndSkillTypeId(Integer housekeeperId, Integer skillTypeId);
}

File: src/main/kotlin/com/itsci/mju/maebanjumpen/housekeeperskill/service/HousekeeperSkillService.kt
package com.itsci.mju.maebanjumpen.housekeeperskill.service

import com.itsci.mju.maebanjumpen.housekeeperskill.dto.HousekeeperSkillDTO
import com.itsci.mju.maebanjumpen.partyrole.dto.HousekeeperDTO
import java.util.Optional

interface HousekeeperSkillService {
    fun getAllHousekeeperSkills(): List<HousekeeperDTO>
    fun getHousekeeperSkillById(id: Int): HousekeeperSkillDTO?
    fun saveHousekeeperSkill(housekeeperSkillDto: HousekeeperSkillDTO): HousekeeperSkillDTO
    fun deleteHousekeeperSkill(id: Int)
    fun getSkillsByHousekeeperId(housekeeperId: Int): Optional<HousekeeperSkillDTO>
    fun updateHousekeeperSkill(id: Int, skillDto: HousekeeperSkillDTO): HousekeeperSkillDTO
    fun updateSkillLevelAndHiresCompleted(housekeeperId: Int, skillTypeId: Int)
    fun findByHousekeeperIdAndSkillTypeId(housekeeperId: Int, skillTypeId: Int): Optional<HousekeeperSkillDTO>
}

